MoreLinq/AggregateRight.cs

```diff
@@ -0,0 +1,129 @@
+        /// </example>
+        /// <remarks>
+        /// This operator executes immediately.
+        /// This operator is the right-associative version of the Aggregate LINQ operator.
+        /// </remarks>
+        public static TAccumulate AggregateRight<TSource, TAccumulate>(this IEnumerable<TSource> source, TAccumulate seed, Func<TSource, TAccumulate, TAccumulate> func)
+        {
+            if (source == null) throw new ArgumentNullException(nameof(source));
+            if (func == null) throw new ArgumentNullException(nameof(func));
+
+            IList<TSource> e = (source as IList<TSource>) ?? source.ToList();
+
+            return AggregateRightImpl(e, seed, func, e.Count);
+        }
+
+        /// <summary>
+        /// Applies a right-associative accumulator function over a sequence.
+        /// The specified seed value is used as the initial accumulator value, 
+        /// and the specified function is used to select the result value.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <typeparam name="TAccumulate">The type of the accumulator value.</typeparam>
+        /// <typeparam name="TResult">The type of the resulting value.</typeparam>
+        /// <param name="source">Source sequence.</param>
+        /// <param name="seed">The initial accumulator value.</param>
+        /// <param name="func">A right-associative accumulator function to be invoked on each element.</param>
+        /// <param name="resultSelector">A function to transform the final accumulator value into the result value.</param>
+        /// <returns>The transformed final accumulator value.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = Enumerable.Range(1, 5);
+        /// int result = numbers.AggregateRight("6", (a, b) => string.Format("({0}/{1})", a, b), str => str.Length);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>21</c>.
+        /// </example>
+        /// <remarks>
+        /// This operator executes immediately.
+        /// This operator is the right-associative version of the Aggregate LINQ operator.
+        /// </remarks>
+        public static TResult AggregateRight<TSource, TAccumulate, TResult>(this IEnumerable<TSource> source, TAccumulate seed, Func<TSource, TAccumulate, TAccumulate> func, Func<TAccumulate, TResult> resultSelector)
+        {
+            if (source == null) throw new ArgumentNullException(nameof(source));
+            if (func == null) throw new ArgumentNullException(nameof(func));
+            if (resultSelector == null) throw new ArgumentNullException(nameof(resultSelector));
+
+            return resultSelector(source.AggregateRight(seed, func));
+        }
+
+        private static TResult AggregateRightImpl<TSource, TResult>(IList<TSource> e, TResult current, Func<TSource, TResult, TResult> func, int i)
```

Comment:
For clarity, rename `e` to `list` and `current` to either `acc` or `accumulator`. `e` is _traditionally_ used for _element_.